mesonbuild/modules/fs.py

```diff
@@ -312,6 +315,23 @@ def copyfile(self, state: ModuleState, args: T.Tuple[FileOrString, T.Optional[st
 
         return ModuleReturnValue(ct, [ct])
 
+    @FeatureNew('fs.relative_to', '1.3.0')
+    @typed_pos_args('fs.relative_to', (str, File, CustomTarget, CustomTargetIndex, BuildTarget), (str, File, CustomTarget, CustomTargetIndex, BuildTarget))
+    @noKwargs
+    def relative_to(self, state: ModuleState, args: T.Tuple[FileOrString, FileOrString], kwargs: TYPE_kwargs) -> str:
+        def to_path(arg: T.Union[FileOrString, CustomTarget, CustomTargetIndex, BuildTarget]) -> str:
+            if isinstance(arg, File):
+                return arg.absolute_path(state.environment.source_dir, state.environment.build_dir)
+            elif isinstance(arg, (CustomTarget, CustomTargetIndex, BuildTarget)):
+                return os.path.join(state.environment.build_dir, state.subdir, arg.get_filename())
```

Comment:
I would use `state.backend.get_target_filename_abs()`. As a bonus it will warn if the customtarget has more than one output.